src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1109,23 +1144,59 @@ void CASESession::SendErrorMsg(SigmaErrorType errorCode)
                    ChipLogError(SecureChannel, "Failed to send error message"));
 }
 
-CHIP_ERROR CASESession::FindValidTrustedRoot(const System::PacketBufferTLVReader & tlvReader, uint32_t nTrustedRoots)
+CHIP_ERROR CASESession::GenerateDestinationID(const ByteSpan & random, const Credentials::CertificateKeyId * trustedRootId,
+                                              NodeId nodeId, MutableByteSpan & destinationId)
 {
-    CertificateKeyId trustedRoot;
-    System::PacketBufferTLVReader suppTlvReader;
-    uint8_t trustedRootId[kTrustedRootIdSize];
+    HKDF_sha_crypto mHKDF;
+    // TODO: read fabricId
+    uint64_t fabricId              = 0;
+    uint16_t destinationMessageLen = kSigmaParamRandomNumberSize + kP256_PublicKey_Length + sizeof(fabricId) + sizeof(nodeId);
```

Comment:
The message is not based on underlying types, but rather on wire data. Suggest using `sizeof(uint64_t)` for FabricID NodeId, especially since if NodeId type changes over time to possibly be wider, the value here will break.

Reply:
Updated to use the existing FabricId and NodeId types instead (which are uint64_t), rather than hardcode it here.